_posts/2022-04-11-1.15-announcement.md

```diff
@@ -0,0 +1,375 @@
+making auto scaling more seamless and complete, by removing side effects of use cases 
+in which different data sources produce varying amounts of data, and - finally - the 
+ability to upgrade SQL jobs without losing the state. By continuing on supporting 
+checkpoints after tasks finished and adding window table valued functions in batch 
+mode, the experience of unified stream and batch processing was once more improved 
+making hybrid use cases way easier. In the SQL space, not only the first step in 
+version upgrades have been added but also JSON functions to make it easier to import 
+and export structured data in SQL. Both will allow users to better rely on Flink SQL 
+for production use cases in the long term. To establish Apache Flink as part of the 
+data processing ecosystem we improved the cloud interoperability and added more sink 
+connectors and formats. And yes we enabled a Scala-free runtime 
+([the hype is real](https://flink.apache.org/2022/02/22/scala-free.html)).
+
+
+## Operating Apache Flink with joy
+
+Even jobs that have been built and tuned by the best engineering teams still need to 
+be operated. Looking at the lifecycle of Flink based projects most of them are built 
+to stay, putting long-term burdens on the people operating them. The many deployment 
+patterns, APIs, tuneable configs, and use cases covered by Apache Flink come at the 
+high cost of support.
+
+
+### Clarifying checkpoint and savepoint semantics
+
+An essential cornerstone of Flink’s fault tolerance strategy is based on 
+[checkpoints](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/ops/state/checkpoints/)
+[and](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/ops/state/checkpoints_vs_savepoints/) 
+[savepoints](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/ops/state/savepoints/). 
+The intention of savepoints has always been to put transitions, 
+backups, and upgrades of Apache Flink jobs in the control of users, checkpoints, on 
+the other hand, are intended to be fully controlled by Flink and guarantee fault 
+tolerance through fast recovery, fail over, etc. Both concepts are quite similar and 
+the underlying implementation also shares the same ideas and some aspects. Still, 
+both concepts have grown apart by following specific feature requests and sometimes 
+neglecting the overarching idea and strategy. It became apparent that this should be 
+aligned and harmonized better. It has been leading to situations in which users have 
+been relying on checkpoints to stop and restart jobs whereas savepoints would have 
+been the right way to go. Also savepoints are fairly slower as they don’t include 
+some of the features that made taking checkpoints so fast. In some cases like 
+resuming from a retained checkpoint in which the checkpoint is somehow considered as 
+a savepoint but it is unclear to the user when they can actually clean it up. To sum 
+it up: users have been confused.
```

Comment:
```suggestion
The purpose of savepoints has always been to put transitions, 
backups, and upgrades of Flink jobs in the control of users. Checkpoints, on 
the other hand, are intended to be fully controlled by Flink and guarantee fault 
tolerance through fast recovery, failover, etc. Both concepts are quite similar and 
the underlying implementation also shares aspects of the same ideas. 

However, both concepts grew apart by following specific feature requests and sometimes 
neglecting the overarching idea and strategy. Based on user feedback, it became apparent that this should be 
aligned and harmonized better and, above all, to make more clear! 

There have been situations in which users relied on checkpoints to stop/restart jobs when savepoints would have 
been the right way to go. It was also not clear that savepoints are slower since they don’t include 
some of the features that make taking checkpoints so fast. In some cases like 
resuming from a retained checkpoint in which the checkpoint is somehow considered a 
a savepoint, it is unclear to the user when they can actually clean it up. 
```